shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Epoch.hs

```diff
@@ -149,27 +128,39 @@ epochTransition = do
   PoolreapState utxoSt' acnt' dstate' pstate'' <-
     trans @(POOLREAP era) $ TRC (pp, PoolreapState utxoSt acnt dstate pstate', e)
 
-  coreNodeQuorum <- liftSTS $ asks quorum
-
-  let pup = proposals . _ppups $ utxoSt'
-  let ppNew = votedValue pup pp (fromIntegral coreNodeQuorum)
-  NewppState utxoSt'' acnt'' pp' <-
-    trans @(NEWPP era) $
-      TRC (NewppEnv dstate' pstate'', NewppState utxoSt' acnt' pp, ppNew)
+  let epochState' =
+        EpochState
+          acnt'
+          ss'
+          (ls {_utxoState = utxoSt', _delegationState = DPState dstate' pstate''})
+          pr
+          pp
+          nm
+
+  UPECState pp' ppupSt' <-
+    trans @(UPEC era) $ TRC (epochState', UPECState pp (_ppups utxoSt'), ())
+  let utxoSt'' = utxoSt' {_ppups = ppupSt'}
+
+  let Coin oblgCurr = obligation pp (_rewards dstate') (_pParams pstate'')
+      Coin oblgNew = obligation pp' (_rewards dstate') (_pParams pstate'')
+      Coin reserves = _reserves acnt'
+      utxoSt''' =
+        assert (_deposited utxoSt'' == Coin oblgCurr) $
```

Comment:
I'll remove this assertion. I just wanted to check that it is true. @JaredCorduan do you know if this is always the case?

Reply:
on the epoch boundary, yes, we expect that to be true